fe/fe-core/src/main/java/org/apache/doris/catalog/Partition.java

```diff
@@ -350,6 +351,11 @@ public boolean visualiseShadowIndex(long shadowIndexId, boolean isBaseIndex) {
 
     @Deprecated
     public static Partition read(DataInput in) throws IOException {
+        if (Env.getCurrentEnvJournalVersion() <= FeMetaVersion.VERSION_129) {
+            Partition partition = new Partition();
+            partition.readFields(in);
+            return partition;
+        }
```

Comment:
I can not figure out code above.

Reply:
Restore job need to read BackupMeta from a file, which is downloaded from repo into a temp dir in FE.
When reading, it specifies the meta version, which is used on the backup doris cluster.

From example, 
we backup data from a 2.1 doris, which meta version is lower or equal than 129.
Then we restore data to a 3.0 cloud doris,
In cloud mode, EnvFactory.getInstance().createPartition() will create a CloudPartition,
this will causes the read function failed.

If we found the meta version here is lower the 129, which means it is absolutely should not return a CloudPartition, We can return a Partition here.